markov_chain.js

```diff
@@ -0,0 +1,92 @@
+'use strict'
+//Create object for Markov Chain
+class MarkovChain {
+  #transitions;
+  #names;
+  #letterCounts;
+
+  constructor() {
+    this.fillArrays();
+  }
+
+  //tested
+  fillArrays() {
+    this.#transitions = this.zeros([27, 27]);
+  }
+
+  zeros(dimensions) {
+    var array = [];
+    for (var i = 0; i < dimensions[0]; ++i) {
+      array.push(dimensions.length == 1 ? 0 : this.zeros(dimensions.slice(1)));
+    }
+    return array;
+  }
+
+  //not tested
+  static charToInt(c) {
+    if(c == " ") {
+      return 26; //blank character is end state for chain
+    }
+    else {
+      return c.toLowerCase().charCodeAt(0) - 97;
+    }
+  }
+
+  //not tested
+  static intToChar(n) {
+    return String.fromCharCode(97 + n);
+  }
+
+  //not tested
+  updateTransitions() {
+    for(let i = 0; i < this.#names.length - 1; i++) {
+      this.update(this.#names[i], this.#names[i + 1]);
+    }
+
+    //last character in list always goes to end state
+    this.update(this.#names[this.#names.length - 1], " ");
+    
+    //convert rows to stochastic vectors
+    for(let i = 0; i < this.#transitions.length; i++) {
+      MarkovChain.normalize(this.#transitions[i]);
+    }
+    console.log(this.#transitions);
+  };
+
+  update(from, to) {
+    this.#transitions[MarkovChain.charToInt(from)][MarkovChain.charToInt(to)]++;
+  }
+
+  static normalize(v) {
+    let sum = v.reduce((a, b) => a + b);
+    for(let i in v) {
+      if(v[i] !== 0) {
+        v[i] /= sum;
+      }  
+    }
+  }
+
+  setNames(s) {
```

Comment:
Either enforce a pattern to follow, or handle the different cases of properly reading in the names